src/analyses/cfg_dominators.h

```diff
@@ -142,96 +293,201 @@ void cfg_dominators_templatet<P, T, post_dom>::initialise(P &program)
   cfg(program);
 }
 
+template <class P, class T, bool post_dom>
+void cfg_dominators_templatet<P, T, post_dom>::assign_postordering(
+  std::size_t start_node_index)
+{
+  struct stack_entryt
+  {
+    // cfg index, not the post-order index we're assigning
+    std::size_t node_index;
+    typename cfgt::nodet::edgest::const_iterator child_iterator;
+    typename cfgt::nodet::edgest::const_iterator child_end;
+  };
+
+  std::size_t next_postorder_index = 0;
+  std::vector<stack_entryt> stack;
+
+  auto place_node_on_stack_if_not_visited = [this,
```

Comment:
Would be good if you can find a way to please clang-format that's less odd than this